client/src/components/Kanban/CardForm/CardForm.tsx

```diff
@@ -0,0 +1,136 @@
+import useColorTagStyles from '../shared/colorStyles';
+import useStyles from './useStyles';
+
+type CardFormProps = {
+  columnId: string;
+};
+
+const CardForm = ({ columnId }: CardFormProps): JSX.Element => {
+  const classes = useStyles();
+  const [formIsOpen, setFormIsOpen] = useState<boolean>(false);
+
+  const openForm = (): void => setFormIsOpen(true);
+  const closeForm = (): void => setFormIsOpen(false);
+
+  const handleClose = (event: globalThis.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      closeForm();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleClose, false);
+    return () => {
+      document.removeEventListener('keydown', handleClose, false);
+    };
+  }, []);
+
+  return (
+    <Box>
+      {formIsOpen ? <InnerForm columnId={columnId} formAction={setFormIsOpen} /> : null}
+      {!formIsOpen && (
+        <Button
+          onClick={openForm}
+          size="large"
+          color="secondary"
+          disableElevation
+          className={`${!formIsOpen ? classes.button : ''}`}
+        >
+          Add a card...
+        </Button>
+      )}
+    </Box>
+  );
+};
+
+type InnerFormProps = {
+  columnId: string;
+  formAction: Dispatch<SetStateAction<boolean>>;
+};
+const InnerForm = ({ columnId, formAction }: InnerFormProps) => {
+  const [name, setName] = useState<string>('');
+  const [selectedTagColor, setTagColor] = useState<string>('white');
+  const { addCard } = useKanban();
+  const theme = useTheme();
+  const classes = useStyles();
+  return (
+    <>
+      <Box className={classes.cardFormWrapper}>
+        <Box className={classes.cardForm}>
+          <InputBase
+            className={classes.input}
+            value={name}
+            onChange={(e) => setName(e.target.value)}
+            placeholder="Add title..."
+          />
+        </Box>
+        <Divider light />
+        <Box className={classes.tagWrapper}>
+          <Grid container alignItems="center" justify="space-between">
+            <Grid item>
+              <Typography className={classes.typography} variant="body1">
+                Select tag:
+              </Typography>
+            </Grid>
+            <Grid item>
+              <Grid container className={classes.colorsWrapper}>
+                {Object.keys(theme.palette.tags).map((tagColor: string): JSX.Element => {
+                  return (
+                    <Color
+                      key={tagColor}
+                      name={tagColor}
+                      activeSelected={tagColor === selectedTagColor}
+                      setSelected={setTagColor}
+                    />
+                  );
+                })}
+              </Grid>
+            </Grid>
+          </Grid>
+        </Box>
+      </Box>
+      <Button
+        onClick={() => {
+          addCard({
+            name,
+            columnId: columnId,
+            id: `card-${Math.floor(Math.random() * 999999)}`,
```

Comment:
try using something like https://www.npmjs.com/package/uuid instead

Reply:
As I mentioned in the meeting, this will eventually turn into an Object ID from Mongo so the likelihood for duplication is rare. Just a temporary measure for right now since we are using placeholder data.